### ISSUE TITLE
Crash when closing query buffers

### ISSUE THREAD

**Issue by jeaye** [rawdigits/wee-slack#41]
Any private message bufffer for slack, when closed using `/close` gives me: http://dpaste.com/2XA2XR8

weechat: 1.1.1
os: Arch x64

This happens around 25% of the time, not every time. If I open up 4 query buffers and try to `/close` them all, I'll end up with the crash.

**Comment by rawdigits** [comment]
still trying to reproduce this one.

can you try commenting out lines #970 #1268 and #1602?

the typing bar item is all i can guess is causing this from the stack trace.

**Comment by jeaye** [comment]
I wasn't able to solve it by commenting out these lines (I did 1610 instead of 1602, since 1602 is a comment). If you have a specific commit/branch you'd like me to check out with your proposed changes (I'm not a python guy), I can do that.

**Comment by jeaye** [comment]
Also, I **don't** have the slack typing notification in my input bar, or any bar, if that helps.

**Comment by jemc** [comment]
I can't see the stack trace in the OP (the dpaste looks like it is expired), but I believe I'm having the same issue.  Here is what I see:

```
*** Very bad! WeeChat is crashing (SIGSEGV received)
*** Full crash dump was saved to /home/jemc/.weechat-cb/weechat_crash_20150315_7069.log file.
***
*** Please help WeeChat developers to fix this bug:
***   1. If you have a core file, please run:  gdb /path/to/weechat core
***      then issue command: "bt full" and send result to developers
***      (see user's guide for more info about report of crashes).
***   2. Otherwise send backtrace (below), only if it is a complete trace.
***      Keep the crash log file, just in case developers ask you some info
***      (be careful, private info like passwords may be in this file).

======= WeeChat backtrace =======
(written by WeeChat 1.1.1, compiled on Feb 25 2015 11:48:59)
001  ??:? [function weechat_backtrace]
002  ??:? [function debug_sigsegv]
003  ??:0 [function ??]
004  ??:0 [function ??]
005  ??:? [function gui_buffer_set_short_name]
006  ??:? [function gui_buffer_set]
007  /home/jemc/1/code/gitx/weechat/src/plugins/python/weechat-python-api.c:? [function weechat_python_api_buffer_set]
008  ??:0 [function ??]
009  ??:0 [function ??]
010  ??:0 [function ??]
011  ??:0 [function ??]
012  ??:0 [function ??]
013  ??:0 [function ??]
014  ??:0 [function ??]
015  ??:? [function weechat_python_exec]
016  ??:? [function weechat_python_api_hook_timer_cb]
017  ??:? [function hook_timer_exec]
018  ??:? [function gui_main_loop]
019  ??:? [function main]
020  ??:0 [function ??]
021  ??:? [function _start]
======= End of  backtrace =======
```

**Comment by jemc** [comment]
So, after commenting out all of the `hook_timer`s in the plugin, I can avoid the SIGSEGV and weechat remains open for me to see the python backtrace:
```
09:55 python: stdout/stderr: Traceback (most recent call last):
09:55 python: stdout/stderr:   File "/home/jemc/.weechat-cb/python/autoload/wee_slack.py", line 1326, in buffer_closing_cb
09:55 python: stdout/stderr:     channels.find(data).closed()
09:55 python: stdout/stderr:   File "/home/jemc/.weechat-cb/python/autoload/wee_slack.py", line 494, in closed
09:55 python: stdout/stderr:     message_cache.pop(self.identifier)
09:55 python: stdout/stderr: KeyError: u'D040V27SR'
09:55 =!= python: error in function "buffer_closing_cb"
```

Wrapping the `message_cache.pop(self.identifier)` line with a try "solves" the issue for me for now:
```python
        try:
            message_cache.pop(self.identifier)
        except KeyError:
            pass
```

Hopefully that helps you track this down.

**Comment by kyrias** [comment]
Just going to leave a quick note that I can reproduce this too, and I'll try @jemc’s workaround later.

**Comment by bchretien** [comment]
:+1: Same here, using the `try/except` workaround for now.

**Comment by rawdigits** [comment]
added your fix in 0.97.23 -- thanks for debugging this!
